habitat/datasets/object_nav/object_nav_dataset.py

```diff
@@ -74,27 +133,17 @@ def from_json(
 
                 episode.scene_id = os.path.join(scenes_dir, episode.scene_id)
 
-            for i in range(len(episode.goals)):
-                episode.goals[i] = ObjectGoal(**episode.goals[i])
-
-                for vidx, view in enumerate(episode.goals[i].view_points):
-                    view_location = ObjectViewLocation(**view)
-                    view_location.agent_state = AgentState(
-                        **view_location.agent_state
-                    )
-                    episode.goals[i].view_points[vidx] = view_location
+            episode.goals = self.goals_by_category[episode.goals_key]
 
             if episode.shortest_paths is not None:
                 for path in episode.shortest_paths:
                     for p_index, point in enumerate(path):
-                        point = {
-                            "action": point,
-                            "rotation": None,
-                            "position": None,
-                        }
+                        if isinstance(point, int) or point is None:
```

Comment:
We switching to action be str.
```suggestion
                        if isinstance(point, int) or point is None or point is str:
```